src/test/java/javax/jmdns/test/DNSRecordTest.java

```diff
@@ -99,4 +100,18 @@ public void testIPv4MappedIPv6Addresses() throws Exception {
         PowerMock.verifyAll();
 
     }
+
+    @Test
+    public void testServiceInfoFromDNSRecord() {
+
+        DNSRecord record = new DNSRecord.Service("test._http._tcp.local.", DNSRecordClass.CLASS_IN, true, TTL_IN_SECONDS, 0, 0, 0, "test_server");
+
+        ServiceInfo serviceInfo = record.getServiceInfo(true);
+
+        Assert.assertEquals("test", serviceInfo.getName());
+        Assert.assertEquals("http", serviceInfo.getApplication());
+        Assert.assertEquals("tcp", serviceInfo.getProtocol());
+        Assert.assertEquals("local", serviceInfo.getDomain());
+        Assert.assertEquals("test_server", serviceInfo.getServer());
```

Comment:
without my fix, `serviceInfo.getServer()` was returning an empty string when instead it was part of the original `record`